code/headers/manual_control.hpp

```diff
@@ -0,0 +1,156 @@
+#include <base_module.hpp>
+#include <controller_interface.hpp>
+
+namespace r2d2::manual_control {
+    class module_c : public base_module_c {
+    private:
+        /**
+         *  \brief
+         *  This function returns the ablosule of a joystick_value_c
+         *
+         *  \details
+         *  This function takes the values from a joystick_value_c and makes
+         * every value positive
+         */
+        joystick_value_c abs(joystick_value_c value) {
+            if (value.x < 0) {
+                value.x *= -1;
+            }
+            if (value.y < 0) {
+                value.y *= -1;
+            }
+            return value;
+        }
+        /**
+         *  \brief
+         *  This function returns the ablosule of an int
+         *
+         *  \details
+         *  This function makes the int positive
+         */
+        int abs(int value) {
+            if (value < 0) {
+                value *= -1;
+            }
+            return value;
+        }
+
+        r2d2::manual_control::controller_interface_c &controller;
+
+        // lists to save previous states of buttons, sliders and joysticks
+        bool button_list[12] = {};
+        unsigned char slider_list[2] = {};
+        joystick_value_c joystick_list[2] = {};
```

Comment:
Again, personally, I would like to see std::arrays

Reply:
Also changed

Reply:
slider_list is still a c style array